crates/database/src/orders.rs

```diff
@@ -716,6 +716,45 @@ pub async fn count_limit_orders_by_owner(
         .await
 }
 
+#[derive(Debug, sqlx::FromRow)]
+pub struct OrderWithQuote {
+    pub order_buy_amount: BigDecimal,
+    pub order_sell_amount: BigDecimal,
+    pub order_fee_amount: BigDecimal,
+    pub quote_buy_amount: BigDecimal,
+    pub quote_sell_amount: BigDecimal,
+    pub quote_gas_amount: f64,
+    pub quote_gas_price: f64,
+    pub quote_sell_token_price: f64,
+}
+
+pub async fn user_orders_with_quote(
```

Comment:
There is already a SQL query that counts limit orders. Instead of fetching all this data and counting in the rust code you could have the in-market check already in SQL to only return a single number.

Reply:
Basically reimplement the logic of `is_order_outside_market_price()` inside `count_limit_orders_by_owner()`. Hopefully there is a way in PSQL to convert the types appropriately for the computation.

Reply:
@MartinquaXD In my opinion, doing arithmetical operations in PSQL query involving U256 and f64 is really not a good idea (unless it is plain sum, and even so), we are going to lose precision, and we are going to push CPU overhead to the database. What's more, I don't think we will save time, since database CPU is less powerful than the backend CPU, and the rust code will be definitely faster. 

I understand your concern of potential long queries in case the user placed many orders, for that please, consider adding the internal class (as I proposed here https://github.com/cowprotocol/services/pull/2456#discussion_r1508995317), which will tackle down all the issues. We could also do the queries by pagination in case of long queries.

Reply:
> not a good idea (unless it is plain sum

Interesting. Why do you think so?
> we are going to lose precision

This is also the case in the rust code since`.fee()`calls `U256::from_f64_lossy()`.

> I don't think we will save time, since database CPU is less powerful than the backend CPU, and the rust code will be definitely faster.

Computations might be faster in rust but I very much doubt that returning all that data, parsing it, allocating the structs (1 Vec per BigDecimal) and doing the computation in rust will be faster than doing a computation inside the db and returning a single number.

This is not a hill I'm willing to die on (although I disagree) if you feel strongly about keeping the computation out of the DB I'd at least consider fetching the data using a stream instead of call `.fetch_all()` to limit the memory consumption when a user has lots of orders.

Reply:
I guess we would need to benchmark this in order to get a definitive answer. It's true that for accounts with a lot of orders this may become a bottleneck. In theory we could also decide the "in-marketness" of an order when it's inserted into the db (and then simply count based on a flag)?

I'd be curious how PSQL math is less accurate than rust math?